### DISCUSSION TITLE
esbuild

### DISCUSSION THREAD

**Discussion by bestickley** [TypeStrong/ts-node#1746] [Q&A] [answered]
Was esbuild considered as a supported transpiler? Why only swc? Coming from Vite where esbuild is used and just curious.

**Comment by cspotcode** [comment, marked as answer]
At the time, swc's performance *for our specific use-case* was better, because swc runs in-process.  At the time, esbuild required running as an external process.  Maybe it still does?  I'm not sure.  Also, *for our specific use-case*, esbuild and swc were functionally identical.

Our transpiler API is public, so anyone can write an esbuild integration.

Is there something specifically where swc is not working?  Because if swc and esbuild are functionally identical *for this specific use-case*, then adding esbuild support is wasted effort.

**Reply by bestickley** [reply]
Thanks for explaining, @cspotcode! swc works great, was just curious.
